File: leetcode/1700-number-of-students-unable-to-eat-lunch.py
# https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/

from collections import Counter
from typing import List

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        counter = Counter(students)

        for sandwiche in sandwiches:
            if counter[sandwiche] == 0:
                return max(counter[0], counter[1])
            else:
                counter[sandwiche] -= 1

        return 0

File: leetcode/1700-number-of-students-unable-to-eat-lunch.js
// https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/

/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function(students, sandwiches) {

  const counter = {'0': 0, '1': 0};

  for (let student of students) {
    counter[student] += 1;
  }

  for (let sandwiche of sandwiches) {
    if (counter[sandwiche] === 0) return Math.max(counter['0'], counter['1']);
    else counter[sandwiche] -= 1;
  }

  return 0;
};
